### ISSUE TITLE
Parse error on type level multiplication

### ISSUE THREAD

**Issue by expipiplus1** [ndmitchell/hlint#227]
Here we go again... :)

I get this with HEAD of hlint.

Interestingly I'm not able to replicate this parse error with `haskell-src-exts-1.17.1`. Is it possible to `hlint` to report a parse error without `haskell-src-exts` failing?

This parses correctly:

``` Haskell
{-# language TypeOperators #-}
type A = B - B
```

However this does not, I assume because of some confusion between the kind `*` and the multiplication operator.

``` Haskell
{-# language TypeOperators #-}
type A = B * B
```

This also parses without a problem

``` Haskell
{-# language TypeOperators #-}
type A = B Prelude.* B
```

**Comment by mpickering** [comment]
Please can you explain what the bug is in HSE? I'm confused about where the actual problem is because of this sentence.

> Interestingly I'm not able to replicate this parse error with haskell-src-exts-1.17.1. Is it possible to hlint to report a parse error without haskell-src-exts failing?

**Comment by ndmitchell** [comment]
You can workaround the bug by passing `-XHaskell98` which disables all extensions, and then just enables the ones you explicitly declare. For HSE, the issue is that the following file fails to parse:

```
{-# language TypeOperators, KindSignatures #-}
type A = B * B
```

But GHC seems to be able to parse it. Since `-XNoKindSignatures` alone doesn't solve it I guess there are other extensions that also enable kind signatures. @mpickering, does that look like an HSE issue?

**Comment by mpickering** [comment]
If there is a file which GHC parses but HSE doesn't then it is an issue.

**Comment by ndmitchell** [comment]
Raised upstream at https://github.com/haskell-suite/haskell-src-exts/issues/298.

**Comment by ndmitchell** [comment]
Fixed in hlint-2.0.15.

**Comment by expipiplus1** [comment]
Thanks!
